docs/modules/rust-guide/examples/profile-tutorial/profile.rs

```diff
@@ -45,22 +51,26 @@ fn update(profile: Profile) {
     profile_store.insert(principal_id, profile);
 }
 
-#[query]
-fn search(text: String) -> Option<&'static Profile> {
+#[query(reply = true)]
+fn search(text: String) -> Empty<Option<Profile>> {
     let text = text.to_lowercase();
     let profile_store = storage::get::<ProfileStore>();
 
+    let mut profile = None;
     for (_, p) in profile_store.iter() {
         if p.name.to_lowercase().contains(&text) || p.description.to_lowercase().contains(&text) {
-            return Some(p);
+            profile = Some(p);
+            break;
         }
 
         for x in p.keywords.iter() {
             if x.to_lowercase() == text {
-                return Some(p);
+                profile = Some(p);
+                break;
```

Comment:
Does this need to break out of the outer `for` loop?

Reply:
Whoops!